app/src/componentsV2/ContentHeader/ContentHeader.tsx

```diff
@@ -1,31 +1,68 @@
-import { Col, Row } from "antd";
-import "./contentHeader.scss";
-
 import React, { ReactNode } from "react";
+import { Button, Col, Input, Row } from "antd";
+import { SearchOutlined } from "@ant-design/icons";
+import { Filter, FilterType } from "./type";
+import "./contentHeader.scss";
 
 export interface Props {
-  title: string;
-  subtitle: string;
-  actions: ReactNode[];
+  title: ReactNode;
+  subtitle: ReactNode;
+  actions?: ReactNode[];
+  searchValue?: string;
+  setSearchValue?: (s: string) => void;
+  filters?: Filter[];
+  activeFilter?: FilterType;
 }
 
 // Contains common design and colors for app
-const ContentHeader: React.FC<Props> = ({ title, subtitle, actions }: Props) => {
+export const ContentHeader: React.FC<Props> = ({
+  title,
+  subtitle,
+  actions,
+  searchValue,
+  setSearchValue,
+  filters = [],
+  activeFilter,
+}) => {
   return (
-    <Row justify={"space-between"} align={"bottom"}>
-      <Col span={12}>
-        <div>{title}</div>
-        <div>{subtitle}</div>
-      </Col>
-      <Col span={12}>
-        <Row align={"bottom"}>
-          {actions.map((action) => {
-            return <Col>{action}</Col>;
-          })}
-        </Row>
-      </Col>
-    </Row>
+    <div>
+      <Row justify="space-between" align="bottom" className="content-header-container">
+        <Col span={12}>
+          <div className="content-header-title">{title}</div>
+          <div className="content-header-subtitle">{subtitle}</div>
+        </Col>
+        <Col className="ml-auto">
+          <Row align="middle" className="content-header-actions">
+            {actions.map((action) => (
+              <Col>{action}</Col>
+            ))}
+          </Row>
+        </Col>
+      </Row>
+      <Row align="middle" className="filters-container">
+        {searchValue !== undefined ? (
+          <Input
+            allowClear
+            prefix={<SearchOutlined />}
+            placeholder="Search by rule name"
+            value={searchValue}
+            onChange={(e) => setSearchValue(e.target.value)}
+            className="search-input"
+          />
+        ) : null}
```

Comment:
Can we give a support to enable search in table or not by passing a prop named `enableSearch` ?

Reply:
We again need to pass `searchValue` along with `enableSearch` prop instead let the content header decide with the `searchValue` only if to show search or not.